pkg/datapath/linux/bandwidth/cell.go

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+//go:build linux
+
+package bandwidth
+
+import (
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/option"
+)
+
+var Cell = cell.Module(
+	"bandwidth-manager",
+	"Linux Bandwidth Manager for EDT-based pacing",
+
+	cell.Config(Config{false, false}),
+	cell.Provide(newBandwidthManager),
+)
+
+type Manager interface {
+	BBREnabled() bool
+	DeleteEndpointBandwidthLimit(epID uint16) error
+	Enabled() bool
+	ResetQueues() bool
+}
+
+type Config struct {
+	// EnableBandwidthManager enables EDT-based pacing
+	EnableBandwidthManager bool
+
+	// EnableBBR enables BBR TCP congestion control for the node including Pods
+	EnableBBR bool
+}
+
+func (def Config) Flags(flags *pflag.FlagSet) {
+	flags.Bool("enable-bandwidth-manager", def.EnableBandwidthManager, "Enable BPF bandwidth manager")
+	flags.Bool(EnableBBR, def.EnableBBR, "Enable BBR for the bandwidth manager")
+}
+
+func newBandwidthManager(lc hive.Lifecycle, p bandwidthManagerParams) Manager {
+	m := &manager{params: p}
+	lc.Append(m)
+	return m
+}
+
+func (m *manager) Start(hive.HookContext) error {
+	err := m.probe()
```

Comment:
Failure to init bw manager could be a good use case for the modular health checker 😉